lib/connection.c

```diff
@@ -645,8 +645,8 @@ char * STDCALL encode_url(CURL *curl,
     // sure extraUrlParams is correct)
     if (extraUrlParams && !is_string_empty(extraUrlParams))
     {
-      strncat(encoded_url, URL_PARAM_DELIM, 1);
-      strncat(encoded_url, extraUrlParams, strlen(extraUrlParams));
+      sb_strncat(encoded_url, encoded_url_size, URL_PARAM_DELIM, 1);
+      sb_strncat(encoded_url, encoded_url_size, extraUrlParams, strlen(extraUrlParams));
```

Comment:
The encoded_url_size in line 617 already calculates the size so this should be good.
617:
encoded_url_size += extraUrlParams ?
                        strlen(extraUrlParams) + strlen(URL_PARAM_DELIM) : 0;